From 2d6f021155b815a3bc9543f6ef82c86303b80643 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 10 Sep 2013 22:26:46 -0500
Subject: [PATCH 183/390] rtl8192du: Fix smatch warning in 
 core/rtw_wlan_util.c

This one fixes a place where & was used, but should have been &&.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_wlan_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rtl8192du/core/rtw_wlan_util.c
+++ b/drivers/staging/rtl8192du/core/rtw_wlan_util.c
@@ -1429,7 +1429,7 @@ void update_capinfo(struct rtw_adapter *
 	/*  Check preamble mode, 2005.01.06, by rcnjko. */
 	/*  Mark to update preamble value forever, 2008.03.18 by lanhsin */
 	/* if (pMgntInfo->RegPreambleMode == PREAMBLE_AUTO) */
-	if (updateCap & shortpreamble) { /*  Short Preamble */
+	if (updateCap && shortpreamble) { /*  Short Preamble */
 		if (pmlmeinfo->preamble_mode != PREAMBLE_SHORT) { /*  PREAMBLE_LONG or PREAMBLE_AUTO */
 			shortpreamble = true;
 			pmlmeinfo->preamble_mode = PREAMBLE_SHORT;
